pkg/processor/metricfrequencyprocessor/metriccache.go

```diff
@@ -22,16 +16,19 @@ type DataPoint struct {
 // metricCache caches data points into two level mapping structure.
 // To easily list all data points of a given metric it keeps a separate cache for each incoming metric.
 type metricCache struct {
+	config *Config
```

Comment:
is there need to keep all config in the structure?

Reply:
Added explicit subconfigs.